### PR TITLE
Fix booking days limit / unavailability periods

### PR DESCRIPTION
**Pull request by pferreir**
Supersedes #1738

### CODE REVIEW

File: indico/modules/rb/controllers/user/reservations.py [hunk 1]
```diff
@@ -375,6 +375,10 @@ def _make_form(self):
                 start_date = datetime.strptime(request.args['start_date'], '%Y-%m-%d').date()
             except ValueError:
                 pass
+            else:
+                if not self._room.check_advance_days(start_date, user=session.user, quiet=True):
```
Comment (ThiefMaster) [inline indico/modules/rb/controllers/user/reservations.py]: elif

Comment (pferreir) [inline indico/modules/rb/controllers/user/reservations.py]: With `try..except`? 😉

Comment (ThiefMaster) [inline indico/modules/rb/controllers/user/reservations.py]: oops :monkey:

File: indico/modules/rb/views/calendar.py [hunk 2]
```diff
@@ -325,14 +349,24 @@ def forReservation(self):
...
+        elif self.nb_period:
+            return {
+                'id': None,
+                'creator': None,
+                'reason': 'Unavailable from {0} until {1}.'.format(format_date(self.nb_period.start_dt),
+                                                                   format_date(self.nb_period.end_dt)),
+                'blocking_url': None,
+                'type': 'nonbookable'
+            }
+        return None
```
Comment (ThiefMaster) [inline indico/modules/rb/views/calendar.py]: i'd move it to an `else:` block to have all the returns on the same level

**Pull request closed without merging**